syncopy/io/load_ft.py

```diff
@@ -0,0 +1,419 @@
+                      varname=filename,
+                      actual="No structure found"
+                      )
+
+    msg = f"Found {len(struct_keys)} structure(s): {struct_keys}"
+    SPYInfo(msg)
+
+    out_dict = {}
+
+    # load only a subset
+    if select_structures is not None:
+        keys = select_structures
+    # load all structures found
+    else:
+        keys = struct_keys
+
+    for skey in keys:
+        if skey not in struct_keys:
+            msg = f"Could not find structure `{skey}` in {filename}"
+            SPYWarning(msg)
+            continue
+
+        structure = struct_container[skey]
+        _check_req_fields(req_fields_raw, structure)
+        data = struct_reader(structure)
+        out_dict[skey] = data
+
+    return out_dict
+
+
+def _read_hdf_structure(h5Group,
+                        h5File,
+                        mem_use,
+                        add_fields=None):
+
+    '''
+    Each Matlab structure contained in
+    a hdf5 MAT-File is a h5py Group object.
+
+    Each key of this Group corresponds to
+    a field in the Matlab structure.
+
+    This is the translation from FT to Syncopy:
+
+    FT     Syncopy
+
+    label - channel
+    trial - trial
+    time  - time
+
+    optional:
+    fsample - samplerate
+
+    cfg - X
+
+    '''
+    # for user info
+    struct_name = h5Group.name[1:]
+
+    # this should be fixed upstream such that
+    # the `defaultDimord` is indeed the default :)
+    AData = AnalogData(dimord=AnalogData._defaultDimord)
+
+    # probably better to define an abstract mapping
+    # if we want to support more FT formats in the future
+
+    # these are numpy arrays holding hdf5 object references
+    # e.i. one per trial, channel, time (per trial)
+    trl_refs = h5Group['trial'][:, 0]
+    time_refs = h5Group['time'][:, 0]
+    chan_refs = h5Group['label'][0, :]
+
+    if 'fsample' in h5Group:
+        AData.samplerate = h5Group['fsample'][0, 0]
+    else:
+        AData.samplerate = _infer_fsample(h5File[time_refs[0]])
+
+    # -- retrieve shape information --
+    nTrials = trl_refs.size
+
+    # peek in 1st trial to determine single trial shape
+    # we only support equal trial lengths at this stage
+    nSamples, nChannels = h5File[trl_refs[0]].shape
+    nTotalSamples = nTrials * nSamples
+
+    itemsize = h5File[trl_refs[0]].dtype.itemsize
+    # in Mbyte
+    trl_size = itemsize * nSamples * nChannels / 1e6
+
+    # assumption: single trial fits into RAM
+    if trl_size > 0.4 * mem_use:
+        msg = f"\nSingle trial is at least 40% of the requested chache size of {mem_use}MB\n"
+        msg += f"Still trying to load {trl_size:.1f}MB trials.."
+        SPYWarning(msg)
+
+    # -- IO process --
+
+    # create new hdf5 dataset for our AnalogData
+    # with the default dimord ['time', 'channel']
+    h5FileOut = h5py.File(AData.filename, mode="w")
+    ADset = h5FileOut.create_dataset("data",
+                                     dtype=np.float32,
```

Comment:
Shouldn't this be `h5File[trl_refs[tk]].dtype` or something like that? Does matlab save everything as `float32`?

Reply:
Nopes, but this magically casts it into `float32` which is our preferred format :)

Reply:
Ah, okay. Maybe we should issue a message (not sure if a warning is actually needed here) that the data is type-cast to float32? Just in the (admittedly rare if even existing) case someone has float64 data that actually depends on 64bit precision.